services/database/user.ts

```diff
@@ -35,4 +48,193 @@ async function getUsers() {
     return users;
 }
 
-export { getUser, getUsers };
+/**
+ * Updates a user with the data corresponding to userInput
+ * @param userInput - data for the updated user
+ * @returns prisma User with updated information
+ */
+
+async function updateUser(userInput: UserInput): Promise<User> {
+    /*
+    Flow:
+    - get role and role data from UserInput
+    - create/update the role record
+    - update the user record
+    - return the user, including the role records
+    */
+    const roleData = userInput.roleData;
+    if (!roleData.id) {
+        roleData.id = userInput.id;
+    }
+    const user = await getUser(userInput.id);
+
+    const updateUserArgs = {
+        data: {
+            firstName: userInput.firstName,
+            lastName: userInput.lastName,
+            role: userInput.role,
+        },
+        where: { id: parseInt(userInput.id) },
+        include: {
+            teacher: true,
+            parent: true,
+            programAdmin: true,
+            volunteer: true,
+        },
+    };
+
+    let updatedUser;
+
+    switch (userInput.role) {
+        case roles.PARENT: {
+            if (user.role && user.role !== roles.PARENT) {
+                return;
+            }
+            [, updatedUser] = await prisma.$transaction([
+                prisma.parent.upsert({
+                    create: roleData,
+                    update: roleData,
+                    where: { id: user.id },
+                }),
+                prisma.user.update(updateUserArgs),
+            ]);
+            break;
+        }
+        case roles.PROGRAM_ADMIN: {
+            if (user.role && user.role !== roles.PROGRAM_ADMIN) {
+                return;
+            }
+            [, updatedUser] = await prisma.$transaction([
+                prisma.programAdmin.upsert({
+                    create: roleData,
+                    update: roleData,
+                    where: { id: user.id },
+                }),
+                prisma.user.update(updateUserArgs),
+            ]);
+            break;
+        }
+        case roles.VOLUNTEER: {
+            if (user.role && user.role !== roles.VOLUNTEER) {
+                return;
+            }
+            [, updatedUser] = await prisma.$transaction([
+                prisma.volunteer.upsert({
+                    create: roleData,
+                    update: roleData,
+                    where: { id: user.id },
+                }),
+                prisma.user.update(updateUserArgs),
+            ]);
+            break;
+        }
+        default: {
+            return;
```

Comment:
Same comment as above, in a future PR, let's return an error message for this case. Can you add some TODO comments just so we can remember? Thanks!